daprdocs/content/en/developing-applications/integrations/cloud-providers/authenticating-azure.md

```diff
@@ -0,0 +1,399 @@
+
+**Authenticating using a PFX certificate:**
+
+| Field | Required | Details | Example |
+|--------|--------|--------|--------|
+| `azureTenantId` | Y | ID of the Azure AD tenant | `"cd4b2887-304c-47e1-b4d5-65447fdd542b"` |
+| `azureClientId` | Y | Client ID (application ID) | `"c7dd251f-811f-4ba2-a905-acd4d3f8f08b"` |
+| `azureCertificate` | One of `azureCertificate` and `azureCertificateFile` | Certificate and private key | `"-----BEGIN PRIVATE KEY-----\n MIIEvgI... \n -----END PRIVATE KEY----- \n -----BEGIN CERTIFICATE----- \n MIICoTC... \n -----END CERTIFICATE-----` |
+| `azureCertificateFile` | One of `azureCertificate` and `azureCertificateFile` | Path to the PFX file containing the certificate and private key | `"/path/to/file.pem"` |
+| `azureCertificatePassword` | N | Password for the certificate if encrypted | `"password"` |
+
+When running on Kubernetes, you can also use references to Kubernetes secrets for any or all of the values above.
+
+**Authenticating with Managed Service Identities (MSI):**
+
+| Field           | Required | Details                    | Example                                  |
+|-----------------|----------|----------------------------|------------------------------------------|
+| `azureClientId` | N        | Client ID (application ID) | `"c7dd251f-811f-4ba2-a905-acd4d3f8f08b"` |
+
+Using MSI you're not required to specify any value, although you may optionally pass `azureClientId` if needed.
+
+### Aliases
+
+For backwards-compatibility reasons, the following values in the metadata are supported as aliases, although their use is discouraged.
+
+| Metadata key               | Aliases (supported but deprecated) |
+|----------------------------|------------------------------------|
+| `azureTenantId`            | `spnTenantId`, `tenantId`          |
+| `azureClientId`            | `spnClientId`, `clientId`          |
+| `azureClientSecret`        | `spnClientSecret`, `clientSecret`  |
+| `azureCertificate`         | `spnCertificate`                   |
+| `azureCertificateFile`     | `spnCertificateFile`               |
+| `azureCertificatePassword` | `spnCertificatePassword`           |
+
+## Generating a new Azure AD application (Service Principal)
+
+To start, create a new Azure AD application which we'll use as Service Principal.
+
+Prerequisites:
+
+- [Azure Subscription](https://azure.microsoft.com/free/)
+- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- [jq](https://stedolan.github.io/jq/download/)
+- OpenSSL (included by default on all Linux and macOS systems, as well as on WSL)
+- The scripts below are optimized for a bash or zsh shell
+
+> If you haven't already, log in to Azure first using the Azure CLI:
+>
+> ```sh
+> # Log in Azure
+> az login
+> # Set your default subscription
+> az account set -s [your subscription id]
+> ```
+
+First, create the Azure AD application with:
```

Comment:
You can oversimplify this by using the `az ad sp create-for-rbac --role Contributor --scopes <optional>`

Reply:
Does Dapr still need an App Registration to authenticate to AAD? if that's the case, ignore above comment.

Reply:
Yes Dapr does need the App Registration to authenticate with AAD. I also would prefer not using the short-hand comment because the SP doesn't need any access at the entire subscription level.